src/hdf5_back_gen.py

```diff
@@ -0,0 +1,496 @@
+#!/usr/bin/env python
+"""This module generates HDF5 backend code found in src/hdf5_back.cc"""
+import json
+import textwrap
+
+with open('../share/dbtypes.json') as f:
+    RAW_TABLE = json.load(f)
+
+V3_TABLE = set(tuple(row) for row in RAW_TABLE[897:])
+
+def list_dep(orig):
+    """Returns a list of dependent dbtypes.
+    
+    Keyword arguments: 
+    orig -- original dbtype.
+    
+    First element of returned list will always be orig itself.
+    """
+    pieces = orig.split("_")
+    if len(pieces) == 1:
+        return pieces
+    containers1 = ["VECTOR", "SET", "LIST", "VL_VECTOR", "VL_SET", "VL_LIST"]
+    containers2 = ["MAP", "PAIR", "VL_MAP", "VL_PAIR"]
+
+    if pieces[0] == "VL":
+        del pieces[0:2]
+    else:
+        del pieces[0]
+
+    for i in range(0, len(pieces)):
+        if pieces[i] == "VL":
+            pieces[i] = pieces[i] + "_" + pieces[i + 1]
+            del pieces[i + 1]
+        if i == len(pieces) - 1:
+            break
+    
+    for i in range(len(pieces) - 1, -1, -1):
+        if pieces[i] in containers1:
+            pieces[i] += "_" + pieces[i + 1]
+            del pieces[i + 1]
+        elif pieces[i] in containers2:
+            pieces[i]+= "_" + pieces[i + 1] + "_" + pieces[i + 2]
+            del pieces[i + 1:i + 3]
+    pieces = [orig] + pieces
+    return pieces
```

Comment:
Ahh yes, it does.  Sorry. I think that the best solution then is to add a column to the dbtypes.json file. This should be similar to the canonical form, but use the dbtype name, instead. Here are some examples:

```
INT -> INT
VL_STRING -> VL_STRING
VECTOR_INT -> [VECTOR INT]
VECTOR_VL_STRING -> [VECTOR, VL_STRING]
VL_MAP_INT_FLOAT -> [VL_MAP, INT, FLOAT]
MAP_INT_VECTOR_FLOAT = [MAP, INT, [VECTOR, FLOAT]]
```

Note that with the template types, the first element of the list is not necessarily a valid type.  This will allow you to look up the appropriate subdbtype. In fact, this might make a better canonical form overall...